### PR TITLE
Add `r-changeforest` package

### PR DESCRIPTION
**Pull request by mlondschien**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- ~~[] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).~~
- [x] If static libraries are linked in, the license of the static library is packaged.
- ~~[ ] Build number is 0.~~
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by mlondschien** [comment]
On windows:
```
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-pc-windows-gnu` target may not be installed
  = help: consider downloading the target with `rustup target add x86_64-pc-windows-gnu`
```

Where is the correct place to add this? `bld.dat`?

File: recipes/r-changeforest/build.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+# from https://github.com/conda-forge/r-randomforest-feedstock
+if [[ $target_platform =~ linux.* ]] || [[ $target_platform == win-32 ]] || [[ $target_platform == win-64 ]] || [[ $target_platform == osx-64 ]]; then
+  export DISABLE_AUTOBREW=1
+  $R CMD INSTALL --build changeforest-r
+else
+  # TODO: What does this do?
+  mkdir -p $PREFIX/lib/R/library/changeforest
+  mv * $PREFIX/lib/R/library/changeforest
+fi
```
Comment (xhochy) [inline recipes/r-changeforest/build.sh:10]: Can you try the more modern approach, please?
```suggestion
export DISABLE_AUTOBREW=1

# shellcheck disable=SC2086
${R} CMD INSTALL --build changeforest-r ${R_ARGS}
```

File: recipes/r-changeforest/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  merge_build_host: true  # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
+# Taken from https://github.com/conda-forge/r-gifski-feedstock
+requirements:
+  build:
+    - rust
```
Comment (xhochy) [inline recipes/r-changeforest/meta.yaml:29]: To support cross-compiling.
```suggestion
    - {{ compiler('rust') }}
```

File: recipes/r-changeforest/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    - {{ compiler('c') }}              # [not win]
```
Comment (xhochy) [inline recipes/r-changeforest/meta.yaml:30]: Might be that you also need the normal c-compiler on Windows, too here. Feel free to skip windows here and we can then have a look in the feedstock to solve this.
```suggestion
    - {{ compiler('c') }}
```

**Comment by mlondschien** [comment]
The windows job fails both for `{{ compiler('rust') }}` and `rust` with
```
error[E0463]: can't find crate for `core`
  |
  = note: the `x86_64-pc-windows-gnu` target may not be installed
  = help: consider downloading the target with `rustup target add x86_64-pc-windows-gnu`
```

For `{{ compiler('rust') }}`, the linux job fails with
```
x86_64-conda-linux-gnu-cc -shared -L$PREFIX/lib/R/lib -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib -L$PREFIX/lib -o changeforest.so entrypoint.o -L./rust/target/release -lchangeforest -L$PREFIX/lib/R/lib -lR
/home/conda/staged-recipes/build_artifacts/r-changeforest_1637924787375/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/9.4.0/../../../../x86_64-conda-linux-gnu/bin/ld: cannot find -lchangeforest
collect2: error: ld returned 1 exit status
make: *** [/home/conda/staged-recipes/build_artifacts/r-changeforest_1637924787375/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/R/share/make/shlib.mk:10: changeforest.so] Error 1
ERROR: compilation failed for package ‘changeforest’
```
For just `rust` it passes.

**Comment by mlondschien** [comment]
> Feel free to skip windows here and we can then have a look in the feedstock to solve this.

I.e. skip builds for windows all-together? These are important to run through for me, so I'd prefer to fix them here if possible.

**Comment by mlondschien** [comment]
This did not work. Recommendations @xhochy @conda-forge/staged-recipes?

**Review by xhochy — APPROVED** [review]

**Pull request merged**